llvm/lib/TargetParser/ARMTargetParser.cpp

```diff
@@ -420,20 +444,35 @@ bool ARM::appendArchExtFeatures(StringRef CPU, ARM::ArchKind AK,
     CPU = "generic";
 
   if (ArchExt == "fp" || ArchExt == "fp.dp") {
+    const ARM::FPUKind DefaultFPU = getDefaultFPU(CPU, AK);
     ARM::FPUKind FPUKind;
     if (ArchExt == "fp.dp") {
+      const bool IsDP = ArgFPUKind != ARM::FK_INVALID &&
+                        ArgFPUKind != ARM::FK_NONE &&
+                        isDoublePrecision(getFPURestriction(ArgFPUKind));
```

Comment:
This function is called in order to set ArgFPUKind, so checking it here (and below) looks strange. I would expect it to always be FK_INVALID, is there any situation in which it's not?

Reply:
The `appendArchExtFeatures` is called repeatedly in `DecodeARMFeatures`. There can be multiple FPU related extensions specified. e.g. `-march=armv8-m.main+nofp+fp+nofp.dp` should disable FPU, enable default FPU(fpv5-d16) and then switch to single precission equivalent FPU(fpv5-sp-d16).

Reply:
I will add a test case for this.